09-schema/00-overview.md

```diff
@@ -5,29 +5,85 @@ permalink: /docs/schema/overview
 ---
 
 ## Why Use a Schema?
+
+Schema isa a mean to address the problems of managing and handling unstructured or loosely structured data.
+
+![Unstructured problems](/docs/images/schema/unstructured-problems.png)
+
+The common problems we encounter when dealing with unstructured or loosely structured data are :
+- The problem of data integrity
+As the data is weakly tied to any particular structure it is hard or even impossible to control the state and validity of the data. As a result we have no guarantees on data consistency and validity.
+- The problem of data accessibility and retrieval
+With the lack of any high-level structure comes the lack of possibility to query the data meaningfully. This is either because our data structure is too low level to express complex queries or that handling the complexity of such queries becomes a problem.
+Consequently we might be forced to ask simple questions only.
+- The problem of data maintenance
+This problem is tied to the integrity problem. Since we have little control over the structure of our data it is hard to maintain either requiring data changes to be carried out with surgical precision or making them difficult since data pollution needs to be taken care of.
+- Starting with loose or no schema only defers the responsibility of schema definition and enforcement in time.
+In production systems we cannot afford not having control over you data. As a result, the schema logic needs to be incorporated at the app level instead of being handled by the database.
+
 A Grakn schema is the blueprint of a Grakn knowledge graph. Using a highly flexible language, we define a schema to model a domain true to nature. Highly interconnected data cannot be stored at scale without an underlying structure - one that is capable of expressing the complexity of the dataset, is easy to understand, and can be extended programmatically, at runtime.
 
-The schema enforces logical integrity and consistency across the dataset. In other words, it ensures that all data always conform to the given structure. Any attempt to add data that violates the schema is disallowed.
+The schema defines a specific, explicit, high-level structure of data that is enforced across the dataset. This allows to provide logical integrity and consistency guarantees of our data. Any attempt to add data not conforming to the defined schema is a schema violation and is not allowed.
```

Comment:
This allows **the database** to provide logical integrity **guarantees** and consistency guarantees ~~of~~ **for** our data
OR
This allows **the database to guarantee that our data has logical integrity and consistency**